diff --git a/enable-threads.js b/enable-threads.js
new file mode 100644
index 0000000..fee6c80
--- /dev/null
+++ b/enable-threads.js
@@ -0,0 +1,73 @@
+/**!
+ * Original file: https://github.com/josephrocca/clip-image-sorter/blob/main/enable-threads.js
+ */
+
+if (typeof window === "undefined") {
+    self.addEventListener("install", () => self.skipWaiting());
+    self.addEventListener("activate", e => e.waitUntil(self.clients.claim()));
+
+    async function handleFetch(request) {
+        if (request.cache === "only-if-cached" && request.mode !== "same-origin") {
+            return;
+        }
+
+        if (request.mode === "no-cors") {
+            request = new Request(request.url, {
+                cache: request.cache,
+                credentials: "omit",
+                headers: request.headers,
+                integrity: request.integrity,
+                destination: request.destination,
+                keepalive: request.keepalive,
+                method: request.method,
+                mode: request.mode,
+                redirect: request.redirect,
+                referrer: request.referrer,
+                referrerPolicy: request.referrerPolicy,
+                signal: request.signal,
+            });
+        }
+
+        let r;
+
+        try {
+            r = await fetch(request);
+        } catch (e) {
+            return new Response();
+        }
+
+        if (r.status === 0) {
+            return r;
+        }
+
+        const headers = new Headers(r.headers);
+        headers.set("Cross-Origin-Embedder-Policy", "credentialless");
+        headers.set("Cross-Origin-Opener-Policy", "same-origin");
+
+        return new Response(r.body, { status: r.status, statusText: r.statusText, headers });
+    }
+
+    self.addEventListener("fetch", function (e) {
+        e.respondWith(handleFetch(e.request));
+    });
+
+} else {
+    (async function () {
+        if (window.crossOriginIsolated !== false) return;
+
+        let registration = await navigator.serviceWorker.register(window.document.currentScript.src).catch(e => console.error("COOP/COEP Service Worker failed to register:", e));
+        if (registration) {
+            console.log("COOP/COEP Service Worker registered", registration.scope);
+
+            registration.addEventListener("updatefound", () => {
+                console.log("Reloading page to make use of updated COOP/COEP Service Worker.");
+                window.location.reload();
+            });
+
+            if (registration.active && !navigator.serviceWorker.controller) {
+                console.log("Reloading page to make use of COOP/COEP Service Worker.");
+                window.location.reload();
+            }
+        }
+    })();
+}
diff --git a/godot.tools.html b/godot.tools.html
index 7299b37..b022bfc 100644
--- a/godot.tools.html
+++ b/godot.tools.html
@@ -409,6 +409,7 @@
 		</div>
 	</div>
 
+	<script type="text/javascript" src="enable-threads.js"></script>
 	<script type="text/javascript" src="godot.tools.js"></script>
 	<script type="text/javascript">//<![CDATA[
 		const progressBar = document.getElementById("progress-bar");
